### PR TITLE
RavenDB-9509 Add the option to change the cache MaxSize in requestExecutor

### PR DESCRIPTION
**Pull request by IdanHaim**

### CODE REVIEW

File: src/Raven.Client/Http/HttpCache.cs [hunk 1]
```diff
@@ -27,9 +26,11 @@ public class HttpCache : IDisposable, ILowMemoryHandler
         /// </summary>
         public int NumberOfItems => _items.Count;
 
+        public long MaxSize { get; set; }
```
Comment (ppekrol) [inline src/Raven.Client/Http/HttpCache.cs:29]: Maybe it would be better to expose it via Conventions?

Comment (IdanHaim) [reply src/Raven.Client/Http/HttpCache.cs:29]: I think we will have to change the HttpCache in requestExecutor if we want to expose it via Conventios. but I am not sure.

Comment (ppekrol) [reply src/Raven.Client/Http/HttpCache.cs:29]: We could introduce MaxHttpCacheSize (or MaxRequestExecutorCacheSize ?) in DocumentConventions and in RequestExecutor constructor initialize the HttpCache with that value. Should be relatively easy to do and the configurability level would be much higher.

Comment (IdanHaim) [reply src/Raven.Client/Http/HttpCache.cs:29]: Done

**Pull request closed without merging**